### PR TITLE
cicd improvements

### PR DESCRIPTION
**Pull request by qindotguan**
Here is a follow-up improvement for https://github.com/vertica/vertica-kubernetes/pull/901

1. If the branch name contains k8s-sync, build-server-legacy will be skipped
2. The admintools related suites in the e2e tests will be skipped to save the job running time
3. add a concurrency group for sync-vcluster job to avoid conflictions
4. add vclsuter commit URL into teams notification-summary

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -661,14 +654,14 @@ jobs:
...
       e2e-leg-10-vcluster,
-      e2e-server-upgrade-admintools,
       e2e-server-upgrade-vcluster,
       e2e-operator-upgrade,
-      e2e-udx-admintools,
       e2e-udx-vcluster
     ]
     runs-on: ubuntu-latest
     if: ${{ always() && !contains(needs.*.result, 'failure') && !contains(needs.*.result, 'cancelled') && contains(github.ref, 'k8s-sync') && (github.event_name == 'push')}} 
+    concurrency:
```
Comment (cchen-vertica) [inline .github/workflows/e2e.yml:663]: Have you tested this one?

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -683,10 +676,27 @@ jobs:
...
+          VCLUSTER_REF=$(echo ${{ github.event.head_commit.message }} | awk '{print $NF}')
+          echo "VCLUSTER_URL=${{ github.server_url }}/vertica/vcluster/commit/$VCLUSTER_REF" >> $GITHUB_ENV
+
+      - name: notification on success
+        if: ${{ always() && contains(needs.sync-vcluster.result, 'success') }}
+        uses: jdcargile/ms-teams-notification@v1.4
+        with:
+          github-token: ${{ github.token }}
+          ms-teams-webhook-uri: ${{ secrets.MS_TEAMS_WEBHOOK_URI }}
+          notification-summary: sync-vcluster job in worflow ${{ github.workflow }} with vcluster commit ${{ evn.VCLUSTER_URL }} on branch ${{ github.ref }} finished with status: ${{ needs.sync-vcluster.result }}
```
Comment (cchen-vertica) [inline .github/workflows/e2e.yml:690]: This message cannot show correctly in the Teams channel. What is "evn" here? Should it be "env"?

File: .github/workflows/e2e.yml [hunk 3]
```diff
@@ -683,10 +676,27 @@ jobs:
...
+          verbose-logging: true
+
+      - name: notification on failure
+        if: ${{ always() && ! contains(needs.sync-vcluster.result, 'success') }}
         uses: jdcargile/ms-teams-notification@v1.4
         with:
           github-token: ${{ github.token }}
           ms-teams-webhook-uri: ${{ secrets.MS_TEAMS_WEBHOOK_URI }}
-          notification-summary: ${{ github.job }} job in worflow ${{ github.workflow }} of ${{ github.repository }} has ${{ job.status }} on branch ${{ github.ref }}
+          notification-summary: sync-vcluster job in worflow ${{ github.workflow }} with vcluster commit ${{ evn.VCLUSTER_URL }} on branch ${{ github.ref }} finished with status: ${{ needs.sync-vcluster.result }}
```
Comment (cchen-vertica) [inline .github/workflows/e2e.yml:700]: Ditto

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (qindotguan) [reply .github/workflows/e2e.yml]: Remove it for now and will put the sync-vcluster job into a queue in the future.

File: .github/workflows/e2e.yml (hunk 2 above)
Comment (qindotguan) [reply .github/workflows/e2e.yml]: Updated the typo together with the format. Now the vcluster commit can be viewed as a link now.

**Review by cchen-vertica — APPROVED** [review]

**Pull request merged**